src/app/core/file/file.service.ts

```diff
@@ -1,18 +1,66 @@
-import { Entity } from "../entity/model/entity";
+import { Entity, EntityConstructor } from "../entity/model/entity";
 import { Observable } from "rxjs";
+import { EntityMapperService } from "../entity/entity-mapper.service";
+import { EntityRegistry } from "../entity/database-entity.decorator";
+import { fileDataType } from "./file-data-type";
+import { filter } from "rxjs/operators";
 
 /**
  * This service allow handles the logic for files/attachments.
  * Files can be uploaded, shown and removed.
  */
 export abstract class FileService {
+  protected constructor(
+    protected entityMapper: EntityMapperService,
+    protected entities: EntityRegistry
+  ) {
+    // TODO maybe registration is to late (only when component is rendered)
+    this.deleteFilesOfDeletedEntities();
```

Comment:
Unfortunately no, it has to happen after the entity schemas are adjusted, which only happens in the subscription to the config service in the app component. So also doing it on config updates doesn't work, because the subscriptions are executed on the wrong order (first subscription is executed first).